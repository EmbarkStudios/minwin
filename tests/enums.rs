mod utils;
use utils::*;

// Ensures we can emit some but not all of a simple enum's values
test!(
    simple,
    [
        BindConfig::Minwin(Default::default()),
        BindConfig::Minwin(MinwinBindConfig {
            enum_style: EnumStyle::Minwin,
            ..Default::default()
        }),
        BindConfig::Minwin(MinwinBindConfig {
            enum_style: EnumStyle::Minwin,
            use_rust_casing: true,
            ..Default::default()
        }),
        BindConfig::Bindgen,
    ],
    [
        "Windows.Win32.Foundation.GENERIC_READ",
        "Windows.Win32.Foundation.GENERIC_WRITE",
        "Windows.Win32.Foundation.GENERIC_ALL",
        "Windows.Win32.Storage.FileSystem.FILE_TYPE_DISK",
        "Windows.Win32.Storage.FileSystem.GetFileType",
        "Windows.Win32.Storage.FileSystem.ClfsMgmtLogWriteNotification",
        "Windows.Win32.Storage.FileSystem.ReadLogNotification",
    ]
);

// Ensures we can emit enums that use core types instead of a typedef
test!(
    core,
    [
        BindConfig::Minwin(Default::default()),
        BindConfig::Minwin(MinwinBindConfig {
            enum_style: EnumStyle::Minwin,
            ..Default::default()
        }),
        BindConfig::Minwin(MinwinBindConfig {
            enum_style: EnumStyle::Minwin,
            use_rust_casing: true,
            ..Default::default()
        }),
        BindConfig::Bindgen,
    ],
    ["Windows.Win32.Foundation.S_OK"]
);

// Ensures we can emit the enum type alias, even if no actual enum variants
// are requested
test!(
    always_emits_alias,
    [
        BindConfig::Minwin(Default::default()),
        BindConfig::Minwin(MinwinBindConfig {
            enum_style: EnumStyle::Minwin,
            ..Default::default()
        }),
        BindConfig::Minwin(MinwinBindConfig {
            enum_style: EnumStyle::Minwin,
            use_rust_casing: true,
            ..Default::default()
        }),
        BindConfig::Bindgen,
    ],
    ["Windows.Win32.UI.Input.Pointer.GetPointerTouchInfo"]
);

// Ensures that enums that clash with handle types are supported
test!(
    handle_and_enum,
    [
        BindConfig::Minwin(Default::default()),
        BindConfig::Minwin(MinwinBindConfig {
            enum_style: EnumStyle::Minwin,
            ..Default::default()
        }),
        BindConfig::Minwin(MinwinBindConfig {
            enum_style: EnumStyle::Minwin,
            use_rust_casing: true,
            ..Default::default()
        }),
        BindConfig::Bindgen,
    ],
    [
        "Windows.Win32.UI.WindowsAndMessaging.SetWindowPos",
        "Windows.Win32.UI.WindowsAndMessaging.GetMessageW",
        "Windows.Win32.UI.WindowsAndMessaging.HWND_NOTOPMOST",
    ]
);
