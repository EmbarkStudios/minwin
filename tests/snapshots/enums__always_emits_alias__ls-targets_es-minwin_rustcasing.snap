---
source: tests/enums.rs
expression: bo.bindings
---
#![allow(non_snake_case, non_upper_case_globals, non_camel_case_types)]
::windows_targets::link!(
    "user32.dll" "system" "GetPointerTouchInfo" fn get_pointer_touch_info(pointer_id :
    u32, touch_info : * mut PointerTouchInfo) -> Bool
);
pub mod PointerButtonChangeType {
    pub type Enum = i32;
}
pub mod PointerFlags {
    pub type Enum = u32;
}
pub mod PointerInputType {
    pub type Enum = i32;
}
pub type Bool = i32;
pub type Handle = isize;
pub type Hwnd = isize;
#[repr(C)]
pub struct Point {
    pub x: i32,
    pub y: i32,
}
#[repr(C)]
pub struct PointerInfo {
    pub pointer_type: PointerInputType::Enum,
    pub pointer_id: u32,
    pub frame_id: u32,
    pub pointer_flags: PointerFlags::Enum,
    pub source_device: Handle,
    pub hwnd_target: Hwnd,
    pub pt_pixel_location: Point,
    pub pt_himetric_location: Point,
    pub pt_pixel_location_raw: Point,
    pub pt_himetric_location_raw: Point,
    pub dw_time: u32,
    pub history_count: u32,
    pub input_data: i32,
    pub dw_key_states: u32,
    pub performance_count: u64,
    pub button_change_type: PointerButtonChangeType::Enum,
}
#[repr(C)]
pub struct PointerTouchInfo {
    pub pointer_info: PointerInfo,
    pub touch_flags: u32,
    pub touch_mask: u32,
    pub rc_contact: Rect,
    pub rc_contact_raw: Rect,
    pub orientation: u32,
    pub pressure: u32,
}
#[repr(C)]
pub struct Rect {
    pub left: i32,
    pub top: i32,
    pub right: i32,
    pub bottom: i32,
}

